QUICKSTART-ECS.md

```diff
@@ -0,0 +1,190 @@
+
+```
+aws ec2 describe-subnets \
+   --region us-west-2 \
+   --filter=Name=vpc-id,Values=VPC_ID \
+   | jq '.Subnets[] | {id: .SubnetId, public: .MapPublicIpOnLaunch, az: .AvailabilityZone}'
+```
+
+You'll want to pick one and save it for the launch command later.
+
+You can choose whether you want public or private.
+* Choose private for production deployments to get maximum isolation of instances.
+* Choose public to more easily debug your instance.
+  These subnets have an Internet Gateway, so if you add a public IP address to your instance, you can talk to it.
+  (You can manually add an Internet Gateway to a private subnet later, so this is a reversible decision.)
+
+Note that if you choose to use the public subnet, you'll need your instance to have a publicly accessible IP address.
+That either means adding `--associate-public-ip-address` to the launch command below, or attaching an Elastic IP address after launch.
+There will be a reminder about this when we talk about the launch command.
+
+Finally, note that if you want to launch in a specific availability zone, make sure you pick a subnet that matches; the AZ is listed right below the public/private status.
+
+### IAM role
+
+The instance we launch needs to be associated with an IAM role that allows for communication with ECS.
+
+ECS provides a
+[managed policy](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs_managed_policies.html#AmazonEC2ContainerServiceforEC2Role)
+with all of the appropriate permissions.
+If you've used ECS before, you may already have an appropriate role in your account called `ecsInstanceRole`.
+If you do not, you can
+[follow the instructions in the ECS Developer Guide to create a role](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/instance_IAM_role.html).
+
+Note down the instance role name in your account for the instructions below.
+
+#### Enabling SSM
+
+If you add SSM permissions, you can use Bottlerocket's default SSM agent to get a shell session on the instance.
+
+To attach the role policy for SSM permissions, run the following (replacing INSTANCE_ROLE_NAME with the name of your instance role):
+
+```
+aws iam attach-role-policy \
+   --role-name INSTANCE_ROLE_NAME \
+   --policy-arn arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+```
+
+Next, to retrieve the instance profile name used to launch instances, run this:
+
+```
+aws iam list-instance-profiles-for-role --role-name INSTANCE_ROLE_NAME --query "InstanceProfiles[*].InstanceProfileName" --output text
+```
+
+Note this down as the INSTANCE_PROFILE_NAME for the final launch command.
+
+### Connecting to your cluster
+
+For the instance to be able to communicate with ECS, we need to make sure to configure the instance with the name of the cluster.
+
+Create a file called `userdata.toml` with the following contents, where CLUSTER_NAME is the name of the cluster you created above (for example, "bottlerocket").
+
+```
+[settings.ecs]
+cluster = CLUSTER_NAME
```

Comment:
I and sanika both landed on the same problem here, "" missing around CLUSTER_NAME and realized Instance was not getting attached to cluster. 
Just a Suggestion:
We can either add a note to validate the .toml file or make this as cluster = "CLUSTER_NAME". Even though example existed above it got neglected.

Reply:
Another Suggestion: A link to addtional userdata [settings](https://github.com/bottlerocket-os/bottlerocket#settings) from Readme.md file.

Reply:
Done, thanks for the suggestions!